src/net_processing.cpp

```diff
@@ -963,6 +963,7 @@ void PeerManagerImpl::RemoveBlockRequest(const uint256& hash)
     state->vBlocksInFlight.erase(list_it);
 
     state->nBlocksInFlight--;
+    m_evictionman.RemoveBlockInFlight(node_id);
```

Comment:
Since the value is checked in the next line, I assume we're going to want `RemoveBlockInFlight` to return its new value.

Reply:
`CNodeState::nBlocksInFlight` is not removed in this PR so i think it would be confusing to also do the check with the value from the eviction manager. This is one of those cases that i didn't like because the variables have to be kept in sync. (Arguably it's pretty easy for this one since `nBlocksInFlight` is only modified in two places)